examples/petsc/areasphere.c

```diff
@@ -0,0 +1,408 @@
+  ierr = VecRestoreArrayRead(coords, &coordArray);
+
+  // Create the vectors that will be needed in setup and apply
+  CeedVector uceed, vceed, qdata;
+  CeedInt nqpts;
+  CeedBasisGetNumQuadraturePoints(basisu, &nqpts);
+  CeedVectorCreate(ceed, qdatasize*nelem*nqpts, &qdata);
+  CeedVectorCreate(ceed, xlsize, &uceed);
+  CeedVectorCreate(ceed, xlsize, &vceed);
+
+  /* Create the Q-function that builds the operator for the geomteric factors
+     (i.e., the quadrature data) */
+  CeedQFunctionCreateInterior(ceed, 1, SetupMassGeo,
+                              SetupMassGeo_loc, &qf_setupgeo);
+  CeedQFunctionAddInput(qf_setupgeo, "x", ncompx, CEED_EVAL_INTERP);
+  CeedQFunctionAddInput(qf_setupgeo, "dx", ncompx*topodim, CEED_EVAL_GRAD);
+  CeedQFunctionAddInput(qf_setupgeo, "weight", 1, CEED_EVAL_WEIGHT);
+  CeedQFunctionAddOutput(qf_setupgeo, "qdata", qdatasize, CEED_EVAL_NONE);
+
+  // Set up the mass operator
+  CeedQFunctionCreateInterior(ceed, 1, Mass, Mass_loc, &qf_apply);
+  CeedQFunctionAddInput(qf_apply, "u", ncompu, CEED_EVAL_INTERP);
+  CeedQFunctionAddInput(qf_apply, "qdata", qdatasize, CEED_EVAL_NONE);
+  CeedQFunctionAddOutput(qf_apply, "v", ncompu, CEED_EVAL_INTERP);
+
+  // Create the operator that builds the quadrature data for the operator
+  CeedOperatorCreate(ceed, qf_setupgeo, NULL, NULL, &op_setupgeo);
+  CeedOperatorSetField(op_setupgeo, "x", Erestrictx, CEED_TRANSPOSE,
+                       basisx, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setupgeo, "dx", Erestrictx, CEED_TRANSPOSE,
+                       basisx, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setupgeo, "weight", Erestrictxi, CEED_NOTRANSPOSE,
+                       basisx, CEED_VECTOR_NONE);
+  CeedOperatorSetField(op_setupgeo, "qdata", Erestrictqdi, CEED_NOTRANSPOSE,
+                       CEED_BASIS_COLLOCATED, CEED_VECTOR_ACTIVE);
+
+  // Create the mass operator
+  CeedOperatorCreate(ceed, qf_apply, NULL, NULL, &op_apply);
+  CeedOperatorSetField(op_apply, "u", Erestrictu, CEED_TRANSPOSE,
+                       basisu, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_apply, "qdata", Erestrictqdi, CEED_NOTRANSPOSE,
+                       CEED_BASIS_COLLOCATED, qdata);
+  CeedOperatorSetField(op_apply, "v", Erestrictu, CEED_TRANSPOSE,
+                       basisu, CEED_VECTOR_ACTIVE);
+
+  // Compute the quadrature data for the mass operator
+  CeedOperatorApply(op_setupgeo, xcoord, qdata, CEED_REQUEST_IMMEDIATE);
+
+  PetscScalar *v;
+  ierr = VecZeroEntries(Vloc); CHKERRQ(ierr);
+  ierr = VecGetArray(Vloc, &v);
+  CeedVectorSetArray(vceed, CEED_MEM_HOST, CEED_USE_POINTER, v);
+
+  // Compute the mesh volume using the mass operator: vol = 1^T \cdot M \cdot 1
+  if (!test_mode) {
+    ierr = PetscPrintf(comm,
+                       "Computing the mesh area using the formula: vol = 1^T M 1\n");
+    CHKERRQ(ierr);
+  }
+
+  // Initialize u and v with ones
+  CeedVectorSetValue(uceed, 1.0);
+  CeedVectorSetValue(vceed, 1.0);
```

Comment:
Why set `vceed` to `1.0` if you are going to overwrite it below?

Reply:
Thanks! I can remove it. I had simply followed the same pattern from `ceed/ex1.c`. Before `CeedVectorSetValue` they were manually initializing all entries of both variables to 1.

Reply:
I can remove it in the other examples, too

Reply:
Yeah, I was about to say that we should.